realm/realm-library/src/androidTest/java/io/realm/RealmResultsTests.java

```diff
@@ -1685,4 +1691,240 @@ public void setValue_specificType_internalNameOnDynamicRealms() {
+        realm.commitTransaction();
+
+        RealmResults<AllTypes> all = realm.where(AllTypes.class)
+                .equalTo("columnString", "alltypes1").findAll();
+        assertEquals(1, all.size());
+        String json = all.asJSON();
+        final String expectedJSON = "[\n" +
+                "    {\n" +
+                "        \"columnString\": \"alltypes1\",\n" +
+                "        \"columnLong\": 1337,\n" +
+                "        \"columnFloat\": 3.1400001,\n" +
+                "        \"columnDouble\": 0.89122999999999997,\n" +
+                "        \"columnBoolean\": false,\n" +
+                "        \"columnDate\": \"" + now + "\",\n" +
+                "        \"columnBinary\": \"010203\",\n" +
+                "        \"columnMutableRealmInteger\": 0,\n" +
+                "        \"columnRealmObject\": [\n" +
+                "            {\n" +
+                "                \"name\": \"dog1\",\n" +
+                "                \"age\": 1,\n" +
+                "                \"height\": 1.1,\n" +
+                "                \"weight\": 10.100000381469727,\n" +
+                "                \"hasTail\": true,\n" +
+                "                \"birthday\": \"" + now + "\",\n" +
+                "                \"owner\": []\n" +
+                "            }\n" +
+                "        ],\n" +
+                "        \"columnRealmList\": [\n" +
+                "            {\n" +
+                "                \"name\": \"dog2\",\n" +
+                "                \"age\": 2,\n" +
+                "                \"height\": 2.0999999,\n" +
+                "                \"weight\": 20.100000381469727,\n" +
+                "                \"hasTail\": false,\n" +
+                "                \"birthday\": \"" + now + "\",\n" +
+                "                \"owner\": []\n" +
+                "            },\n" +
+                "            {\n" +
+                "                \"name\": \"dog3\",\n" +
+                "                \"age\": 3,\n" +
+                "                \"height\": 3.0999999,\n" +
+                "                \"weight\": 30.100000381469727,\n" +
+                "                \"hasTail\": true,\n" +
+                "                \"birthday\": \"" + now + "\",\n" +
+                "                \"owner\": [\n" +
+                "                    {\n" +
+                "                        \"name\": \"Dog owner 1\",\n" +
+                "                        \"dogs\": [],\n" +
+                "                        \"cat\": []\n" +
+                "                    }\n" +
+                "                ]\n" +
+                "            }\n" +
+                "        ],\n" +
+                "        \"columnStringList\": [\n" +
+                "            {\n" +
+                "                \"!ARRAY_VALUE\": \"Foo\"\n" +
```

Comment:
It seems strange that primitive lists are modeled like `[{ "!ARRAY_VALUE": "Foo"}]` instead of `["Foo"]`. It would be much more natural to do the later?

I do realize there is a time constraint here, but I would much prefer to have the best possible output if we are adding this as a public API. 

Alternatively, we could expose it as a package protected method until we get a chance to fix this?